rust/user-info-fetcher/src/main.rs

```diff
@@ -0,0 +1,262 @@
+use std::{
+    collections::HashMap,
+    net::AddrParseError,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use axum::{extract::State, routing::post, Json, Router};
+use clap::Parser;
+use futures::{future, pin_mut, FutureExt};
+use moka::future::Cache;
+use reqwest::ClientBuilder;
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use stackable_opa_crd::user_info_fetcher as crd;
+use tokio::{fs::File, io::AsyncReadExt};
+
+mod backend;
+mod http_error;
+mod util;
+
+pub const APP_NAME: &str = "opa-user-info-fetcher";
+
+#[derive(clap::Parser)]
+pub struct Args {
+    #[clap(long, env)]
+    config: PathBuf,
+    #[clap(long, env)]
+    credentials_dir: PathBuf,
+    #[clap(flatten)]
+    common: stackable_operator::cli::ProductOperatorRun,
+}
+
+#[derive(Debug, Clone)]
+struct AppState {
+    config: Arc<crd::Config>,
+    http: reqwest::Client,
+    credentials: Arc<Credentials>,
+    user_info_cache: Cache<UserInfoRequest, UserInfo>,
+}
+
+#[derive(Debug)]
```

Comment:
Ideally this shouldn't be `Debug`, or it should at least stub out `secret`.

Reply:
I believe it was needed for Deserializing.

Reply:
Fixed in cd7920d